v2client/src/main/java/com/yubico/client/v2/VerificationRequester.java

```diff
@@ -171,17 +184,37 @@ public VerificationResponse call() throws Exception {
 			try {
 				return new VerificationResponseImpl(getResponseStream(url));
 			} catch (IOException e) {
-				log.warn("Exception when requesting {}: {}", url.getHost(), e.getMessage());
+				log.warn("Exception when requesting {}.", url.getHost(), e);
 				throw e;
 			}
 		}
 
 		protected InputStream getResponseStream(URL url) throws IOException {
-			HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-			conn.setRequestProperty("User-Agent", userAgent);
-			conn.setConnectTimeout(15000);
-			conn.setReadTimeout(15000);
-			return conn.getInputStream();
+			int attempt = 0;
+			IOException lastException;
+
+			do {
+				try {
+					HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+					conn.setRequestProperty("User-Agent", userAgent);
+					conn.setConnectTimeout(15000);
+					conn.setReadTimeout(15000);
+					return conn.getInputStream();
+				} catch (IOException e) {
+					log.warn("Exception when requesting {}, retrying.", url.getHost(), e);
+					lastException = e;
+				}
+
+				try {
+					// Delay a little bit and hope whatever is happening network-wise clears up.
+					Thread.sleep(500);
+				} catch (InterruptedException e) {
+					// Oh well. Try again anyway.
+				}
+				attempt++;
+			} while (attempt < maxRetries + 1);
```

Comment:
This would run into issues if `maxRetries = Integer.MAX_VALUE`, wouldn't it? Can we use `<=` instead to get around that?
```suggestion
			} while (attempt <= maxRetries);
```

On that note, we might also want to throw an `IllegalArgumentException` for negative values in `YubicoClient.setMaxRetries()`. They would "work" fine with this implementation, but I see no situation where you'd intentionally set a negative value.